feat :

MaXM91 opened the pull request:
- ввести возможность удаления фильмов и юзеров по ид.
-  Добавить методы: 
    - deleteUserById в UserController.java, 
    - deleteById в UserService.java, 
    - deleteFilm в FilmController.java, 
    - deleteById в FilmService.java.
- провести тестирование методов. 
- Добавить методы:
      - shouldDeleteFilmById() в FilmRepositoryTests.java,
      - shouldDeleteUserById() в UserRepositoryTests.java.

Commit:
feat :
- ввести возможность удаления фильмов и юзеров по ид.
    Добавить методы:
    - deleteUserById в UserController.java,
    - deleteById в UserService.java,
    - deleteFilm в FilmController.java,
    - deleteById в FilmService.java.
- провести тестирование методов.
    Добавить методы:
    - shouldDeleteFilmById() в FilmRepositoryTests.java,
    - shouldDeleteUserById() в UserRepositoryTests.java.

Commit:
feat :
- ввести возможность удаления фильмов и юзеров по ид.
    Добавить методы:
    - deleteUserById в UserController.java,
    - deleteById в UserService.java,
    - deleteFilm в FilmController.java,
    - deleteById в FilmService.java.
- провести тестирование методов.
    Добавить методы:
    - shouldDeleteFilmById() в FilmRepositoryTests.java,
    - shouldDeleteUserById() в UserRepositoryTests.java.

Commit:
feat :
- ввести возможность удаления фильмов и юзеров по ид.
    Добавить методы:
    - deleteUserById в UserController.java,
    - deleteById в UserService.java,
    - deleteFilm в FilmController.java,
    - deleteById в FilmService.java.
- провести тестирование методов.
    Добавить методы:
    - shouldDeleteFilmById() в FilmRepositoryTests.java,
    - shouldDeleteUserById() в UserRepositoryTests.java.

Commit:
feat :
- ввести возможность удаления фильмов и юзеров по ид.
    Добавить методы:
    - deleteUserById в UserController.java,
    - deleteById в UserService.java,
    - deleteFilm в FilmController.java,
    - deleteById в FilmService.java.
- провести тестирование методов.
    Добавить методы:
    - shouldDeleteFilmById() в FilmRepositoryTests.java,
    - shouldDeleteUserById() в UserRepositoryTests.java.

CishyF commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -64,6 +64,15 @@ public Film updateFilm(@Valid @RequestBody Film film) {
         return updatedFilm;
     }
 
+    @DeleteMapping("/{filmId}/")
```
По REST в конце пути не должно быть слэшей

MaXM91 replied:
Зачем ревьюить уже правленное?

CishyF commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -64,6 +64,15 @@ public Film updateFilm(@Valid @RequestBody Film film) {
 
+    @DeleteMapping("/{filmId}/")
+    public Film deleteFilm(@PathVariable int filmId) {
+        log.info("Пришел Delete-запрос /films с телом={}", filmId);
```
Напиши DELETE большими буквами пожалуйста. Еще filmId это не тело, а переменная пути, все контроллеры приведены к единой стилистике, чтобы их было удобно читать, я располагал все методы в порядке GET -> POST -> PUT -> DELETE. Мне бы хотелось, чтобы все так и оставалось

MaXM91 replied:
исправлено

CishyF commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -64,6 +64,15 @@ public Film updateFilm(@Valid @RequestBody Film film) {
+    public Film deleteFilm(@PathVariable int filmId) {
+        log.info("Пришел Delete-запрос /films с телом={}", filmId);
+
+        Film deletedFilm = filmService.deleteById(filmId);
```
Запрос на удаление чаще всего не имеет тела ответа

MaXM91 replied:
Проверки постмана прошли. Исправлено.

CishyF commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -73,6 +73,15 @@ public User updateUser(@Valid @RequestBody User user) {
+        log.info("Пришел Delete-запрос /users/{}", userId);
+
+        User deletedUser = userService.deleteById(userId);
+        log.info("Пользователь user={} успешно обновлен", deletedUser);
```
Все-таки удален, а не обновлен

MaXM91 replied:
Исправлено

Commit:
feat :
- ввести возможность удаления фильмов и юзеров по ид.
    Добавить методы:
    - deleteUserById в UserController.java,
    - deleteById в UserService.java,
    - deleteFilm в FilmController.java,
    - deleteById в FilmService.java.
- провести тестирование методов.
    Добавить методы:
    - shouldDeleteFilmById() в FilmRepositoryTests.java,
    - shouldDeleteUserById() в UserRepositoryTests.java.
